erts/emulator/beam/hash.c

```diff
@@ -67,7 +67,7 @@ void hash_get_info(HashInfo *hi, Hash *h)
     (void)objects;
 
     hi->name  = h->name;
-    hi->size  = hash_get_slots(h);
+    hi->size  = size;
```

Comment:
Should be safe to do, but are you sure that gcc/clang just don't optmize this away anyway? I don't mind this change anyway as it adds a bit of clarity.